README.md

```diff
@@ -615,6 +616,29 @@ elao_enum:
 Beware that your database platform must support it. Also, the Doctrine diff tool is unable to detect new or removed 
 values, so you'll have to handle this in a migration yourself.
 
+### Using Enum objects with QueryBuilder
+
+When using enum objects as parameters in a query made with `Doctrine\ORM\QueryBuilder`, the enum objects are cast to database values using the `__toString()` method as the parameter type can not be inferred correctly. To allow the query builder to cast the object to the database value correctly, pass the registered DBAL type as the 3rd parameter in `setParameter()`.
+
+```php
+class Card
+{
+  /**
+   * @ORM\Column(type="suit", length=255)
+   */
+  protected ?SuitEnum $suitType = null;
+}
+
+```
+
+```php
+$em->createQueryBuilder()
+  ->select('c')
+  ->from('Card', 'c')
+  ->where('c.suitType = :param1')
+  ->setParameter('param1', SuitEnum::get(SuitEnum::SPADES), 'suit');
```

Comment:
This is v1 so it's either:
```suggestion
  ->setParameter('param1', SuitEnum::SPADES);
```

or:
```suggestion
  ->setParameter('param1', SuitEnum::SPADES(), 'suit');
```

(using the Doctrine type to convert it to right value).

Reply:
Oh right, I missed this was targeting the 1.x branch.

Reply:
Maybe the example is not the best. I agree that with static values, using the actual constant is the most strait forward and probably preferred choice.

When using a return value which is either null or an enum will not always work correctly if the 3rd parameter is not set. This is what my PR tries to explain. Any suggestions on how to best put this in the docs?

The problem is that if the return value of the function is an enum, the query does not fail and no errors are given. It will use __toString() and (generally) use the incorrect value in the actual query.

```php
$em->createQueryBuilder()
  ->select('c')
  ->from('Card', 'c')
  ->where('c.suitType = :param1')
  ->setParameter('param1', $filter->getSuitType(), 'suit');
```

Reply:
What about:

```php
$em->createQueryBuilder()
  ->select('c')
  ->from('Card', 'c')
  ->where('c.suitType = :param1')
  ->setParameter('param1', $filter->getSuitType()?->getValue());
  // or, PHP < 8
  ->setParameter('param1', $filter->getSuitType() ? $filter->getSuitType()->getValue() : null);
```

? But you can include your third syntax as well and let the users choose what they prefer.